src/librustc/hir/def.rs

```diff
@@ -53,13 +53,13 @@ pub enum Def {
     Existential(DefId),
     /// `type Foo = Bar;`
     TyAlias(DefId),
-    TyForeign(DefId),
+    Foreign(DefId),
     TraitAlias(DefId),
     AssociatedTy(DefId),
     /// `existential type Foo: Bar;`
     AssociatedExistential(DefId),
     PrimTy(hir::PrimTy),
```

Comment:
Hm, maybe drop `Ty` here too? And in the other cases perhaps?

Reply:
No, this is "primitive type". `Def` contains both types and values.